Common/src/main/java/com/blamejared/crafttweaker/impl/registry/zencode/ZenClassRegistry.java

```
@@ -0,0 +1,449 @@
+            
+            this(
+                    Collections.unmodifiableList(view.registeredClasses()),
+                    Maps.unmodifiableBiMap(view.globals()),
+                    Maps.unmodifiableBiMap(view.classes()),
+                    Multimaps.unmodifiableMultimap(view.expansions())
+            );
+        }
+        
+        private void inheritFrom(final ClassData other) {
+            
+            other.registeredClasses().forEach(clazz -> {
+                if(this.registeredClasses().contains(clazz)) {
+                    throw new IllegalStateException("Unable to register the same class twice: " + clazz.getName());
+                }
+                this.registeredClasses().add(clazz);
+            });
+            other.globals().forEach((global, clazz) -> {
+                if(this.globals().containsKey(global) || this.globals().containsValue(clazz)) {
+                    throw new IllegalStateException("Unable to register a global twice: (" + global + ", " + clazz + ')');
+                }
+                this.globals().put(global, clazz);
+            });
+            other.classes().forEach((name, clazz) -> {
+                if(this.classes().containsKey(name) || this.classes().containsValue(clazz)) {
+                    throw new IllegalStateException("Unable to register a class twice: (" + name + ", " + clazz + ')');
+                }
+                this.classes().put(name, clazz);
+            });
+            other.expansions().forEach((target, expansion) -> {
+                final Collection<Class<?>> current = this.expansions().get(target);
+                if(current.contains(expansion)) {
+                    throw new IllegalStateException("Unable to register an expansion for " + target + " twice: " + expansion);
+                }
+                this.expansions().put(target, expansion);
+            });
+        }
+        
+    }
+    
+    private static final class TypeVerifier {
```

Do 80 lines of Code not warrant a separate file?

---

This is mostly a stop-gap as I do not know if there's a better way to check this stuff that also does not swallow exceptions: IIRC the ZenCode parse method swallows.